blocks/shop/shop.php

```diff
@@ -0,0 +1,232 @@
+<?php
+/**
+ * Plugin Name:       Shop
+ * Description:       Dynamic Block which renders a block of Shop Catalog Products.
+ * Requires at least: 5.8
+ * Requires PHP:      7.0
+ * Version:           0.1.0-beta04
+ * Author:            The WordPress Contributors
+ * License:           GPL-2.0-or-later
+ * License URI:       https://www.gnu.org/licenses/gpl-2.0.html
+ * Text Domain:       cata
+ *
+ * @package           Cata\Blocks
+ */
+
+namespace Cata\Blocks;
+
+use stdClass;
+
+/**
+ * Registers the block using the metadata loaded from the `block.json` file.
+ * Behind the scenes, it registers also all assets so they can be enqueued
+ * through the block editor in the corresponding context.
+ *
+ * @see https://developer.wordpress.org/reference/functions/register_block_type/
+ */
+function create_block_shop_block_init() {
+	register_block_type( __DIR__ . '/build', array(
+		'render_callback' => __NAMESPACE__ . '\\render_callback',
+	) );
+}
+add_action( 'init', __NAMESPACE__ . '\\create_block_shop_block_init' );
+
+/**
+ * Products Block Render Callback
+ * 
+ * @param array $block_attributes
+ * @return string
+ */
+function render_callback( $block_attributes) {
+	$url_hash = "";
+	$decoded_results = [];
+	$transient = "";
+
+	if ( empty( $block_attributes['query_url'] )  ) {
+		return;
+	} else {
+		$url_hash = wp_hash( $block_attributes['query_url'] );
+	}
+
+	if ( empty( $url_hash ) ) {
+		return;
+	} else {
+		$transient = get_transient( "cata-blocks-shop-proxy-req-result-" . $url_hash );
+	}
+
+	if ( empty( $transient ) ) {
+		return;
+	} else {
+		$decoded_results = json_decode( $transient );
+	}
+
+	if ( empty( $decoded_results ) ) {
+		return;
+	}
+
+	$products = array_map(
+		__NAMESPACE__ . '\\render_product',
+		$decoded_results
+	);
+
+	$products_string = implode( "\n", $products );
+
+	// TO-DO: change all classnames/rules to ...-shop from ...-products
+	$classnames = get_classnames('wp-block-cata-products wp-block-cata-shop', $block_attributes);
+
+	return "<div class=\"${classnames}\">
+		<div class=\"wp-block-cata-products__layout\">${products_string}</div>
+	</div>";
+}
+
+/**
+ * Render Product
+ * 
+ * @param stdClass $product
+ * @return string
+ */
+function render_product( stdClass $product ) : string {
+	$link   = esc_url( apply_filters( 'creepycatalog_product_block_link', $product->permalink ) );
```

Comment:
This filter name should probably change to cata_ from creepycatalog_ and use whatever block name we decide on.

Reply:
Ok, so I added the `xxx_add_product_link_utm` filter and callback to the shop-block from CC into the .cata-blocks.php` file (for now, at least). The callback and filter names are `cata_` prefixed as well now: `cata_product_block_link`, `cata_add_product_link_utm`.

I only needed to modify the part of the code which adds the query arg for `utm_campaign` to the link url. I simply added this line and set the `utm_campaign` query arg equal to the value of this:
```php
$site_name = strtolower( preg_replace("/\s+/", "", get_bloginfo( 'show' ) ) );
```
So, when testing this on CC locally, before adding the `strtolower()` and the `preg_replace()`, the return value of `get_bloginfo( 'show' )` was `Creepy Catalog` which ended up in the URL just like that. After, the output in the product link was `creepycatalog`, as expected.